src/Tribe/Widget/Widget_Abstract.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+namespace Tribe\Widget;
+
+use Tribe__Utils__Array as Arr;
+
+/**
+ * The abstract base without Views that all widgets should implement.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Widget
+ */
+abstract class Widget_Abstract extends \WP_Widget implements Widget_Interface {
+
+	/**
+	 * Slug of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $slug;
+
+	/**
+	 * Default arguments to be merged into final arguments of the widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,mixed>
+	 */
+	protected $default_arguments = [];
+
+	/**
+	 * Array map allowing aliased widget arguments.
+	 *
+	 * The array keys are aliases of the array values (i.e. the "real" widget attributes to parse).
+	 * Example array: [ 'alias' => 'canonical', 'from' => 'to', 'that' => 'becomes_this' ]
+	 * Example widget usage: [some_tag alias=17 to='Fred'] will be parsed as [some_tag canonical=17 to='Fred']
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,string>
+	 */
+	protected $aliased_arguments = [];
+
+	/**
+	 * Array of callbacks for validation of arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,callable>
+	 */
+	protected $validate_arguments_map = [];
+
+	/**
+	 * Arguments of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,mixed>
+	 */
+	protected $arguments;
+
+	/**
+	 * HTML content of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $content;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function __construct( $id_base = '', $name = '', $widget_options = [], $control_options = [] ) {
+		$arguments = $this->get_arguments();
+
+		parent::__construct(
+			Arr::get( $arguments, 'id_base', '' ),
+			Arr::get( $arguments, 'name', '' ),
+			Arr::get( $arguments, 'widget_options', [] ),
+			Arr::get( $arguments, 'control_options', [] )
+		);
+
+		// @todo add what this does in in TEC-3612 & TEC-3613.
+		$this->setup();
+	}
+
+	/**
+	 * Setup the widget.
+	 *
+	 * @todo update in TEC-3612 & TEC-3613
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	public abstract function setup();
+
+	/**
+	 * Echoes the widget content.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $args     Display arguments including 'before_title', 'after_title',
+	 *                                      'before_widget', and 'after_widget'.
+	 * @param array<string,mixed> $instance The settings for the particular instance of the widget.
+	 */
+	public function widget( $args, $instance ) {
+		echo $this->get_html();
```

Comment:
Do we not need to pass in $args?

Reply:
Or maybe set `$this->arguments = array_merge( $this->arguments, $args )` or have a helper method to do that

Reply:
I think this is to satisfy the code from WP, I am not sure. the arguments are passed on initialization from what I understand of the code,  it doesn't change later on.

Reply:
The widget args are passed from the widget() function, the widget class is not instanced with the current widget's info until the widget() function call itself.

Reply:
`$args` contains widget-specific args.

`$instance` contains the current instance's widget options, things that would be displayed and saved from the widget form itself that the current widget offers.

Neither are getting set for further reference by this method right now, that's what I'm drawing attention to here.

Reply:
You are correct Scott, they are not referenced. The widget(), form(), and update() method are placeholders for this PR and not developed yet. This can only produce a basic widget with no options. TEC-3612 & TEC-3613 will further develop these methods. Along with TEC-3620 & TEC-3597 which connects the backend data to the frontend templates.

Reply:
Let's signal that intent with a comment or a todo comment if possible. I thought this was a bug :)

All good with doing this later.

Reply:
Add the todos about the methods that will be completed in other tickets.